File: Go_Learning/61P_Spiral_Print.go
package main

import (
    "fmt"
)


func SpiralPrint(ekahs [][]int) {

    for _, anim := range ekahs {
            fmt.Println(anim)
    }

    top, bottom, left, right := 0, len(ekahs) - 1, 0, len(ekahs) - 1
    fmt.Println(left, right, top, bottom)
    fmt.Println("\n\n")
    // for anim := range (len(ekahs)) {
    //     for bnim := range(len(ekahs[anim])){
    //         fmt.Println(ekahs[anim][bnim])
    //     }
    // }

    for left <= right && top <= bottom {

        for anim := left; anim <= right; anim ++ {
            fmt.Println(ekahs[top][anim])
        }
        top ++

        for anim := top; anim <= bottom; anim++ {
            fmt.Println(ekahs[anim][right])
        }
        right--

        if top <= bottom {
            for anim := right; anim >= left; anim-- {
                fmt.Println(ekahs[bottom][anim])
            }
            bottom--
        }

        // Print the leftmost column (in reverse)
        if left <= right {
            for anim := bottom; anim >= top; anim-- {
                fmt.Println(ekahs[anim][left])
            }
            left++
        }

    }
    
}



func main() {

    exp := [][]int{
                    {1, 2, 3},
                    {4, 5, 6}, 
                    {7, 8, 9},
    }

    SpiralPrint(exp)


}

File: 61P_Spiral_Print.py
class OmniCreate:
    def arraySpiral_Print(self, ekahs):

        print(ekahs)
        print("\n"*3)
        output = None
        top, bottom, left, right = 0, len(ekahs) -1 , 0, len(ekahs) - 1


        while top <= bottom and left <= right:

            # Travel on 0th index of overall array till last
            for anim in range(left, right + 1):
                print(ekahs[top][anim])
            
            top += 1

             # go down printing last within index till last overall array

            for anim in range(top, bottom + 1):
                print(ekahs[anim][right])

            right -= 1

            # print(left, right, top, bottom)

            # if top <= bottom:
            for anim in range(right, left - 1, -1):
                print(ekahs[bottom][anim])
            bottom -= 1

            # Print the leftmost column (in reverse)
            # if left <= right:
            for anim in range(bottom, top - 1, -1):
                print(ekahs[anim][left])
            left += 1



        return None


create = OmniCreate()

create.arraySpiral_Print([[1, 2, 3],
                          [4, 5, 6], 
                          [7, 8, 9]])
